### PR TITLE
Interactive notifications for windows

### PR DESCRIPTION
**Pull request by Araxeus**
----

> Could be changed to 2nd option which might be better:

Unicode: [`\u{2770}`-previous]  [`\u{25B6}`-play] [`\u{2771}`-next]
(would need a subtle change of code since output for that play button [뿯▽] isn't the same as input for some reason)

----

The same lib could be used to setup interactive notification for mac, if anyone wants to work on that - [Node Notifier](https://www.npmjs.com/package/node-notifier)

- The standard pause button unicode wasn't used, because windows render those media buttons as blue

- Because of a bug with the delivery method, appID cannot be set without breaking the button functionality 
 (it makes returned `data` equal `undefined`)
so appID will it will always be the default "SnoreToast".

 - Song Icon is resized and cropped to keep original aspect ratio (resizing was really ugly with wide images before)

- Made `Notification Priority` menu item show only on linux (since it works only on linux)

- SongInfoProvider was parsing artist name based on video request , not on song - resulting in buggy artist names.
  changed it scrape artist name from playbar (native artist name)
  - could move functionality to notifications plugin instead of infoProvider
  - could be tweaked to wait a second on first run (first song always returns old VideoArtistName because playBar hasn't loaded by the time it queries it)

- `win.once()` instead of `win.on` fix multiple callbacks being registered

### CODE REVIEW

File: plugins/notifications/back.js [hunk 1]
```diff
@@ -39,13 +43,17 @@ module.exports = (win, options) => {
 				}
 				return;
 			}
-			// If url isn't the same as last one - send notification
+			// If url isn"t the same as last one - send notification
```
Comment (th-ch) [inline plugins/notifications/back.js]: nit: typo?

File: plugins/notifications/interactive.js [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    onPause = unpauseNotification;
+
+    win.webContents.once("closed", () => {
+        Delete()
+    });
+}
+
+//delete old notification
+let toDelete;
+function Delete() {
```
Comment (th-ch) [inline plugins/notifications/interactive.js]: nit: functions are generally written lower-case, capital case being kept for classes
```suggestion
function deleteNotification() {
```

**Review by th-ch — APPROVED** [review]
Looks ok, thanks for the contribution! 2 small nits (feel free to address them), otherwise it should be good to merge!

File: plugins/notifications/back.js (hunk 1 above)
Comment (Araxeus) [reply plugins/notifications/back.js]: oh woupsie it must have happened when I replaced all `'` with `"` 😅

File: plugins/notifications/interactive.js (hunk 2 above)
Comment (Araxeus) [reply plugins/notifications/interactive.js]: missed this one, thanks :)

**Review by th-ch — APPROVED** [review]
Looks good, thanks for the contribution!

**Pull request merged**